src/move/emojicoin_dot_fun/sources/emojicoin_dot_fun.move

```diff
@@ -386,6 +401,27 @@ module emojicoin_dot_fun::emojicoin_dot_fun {
             market_address,
         );
 
+        // Verify all emoji bytes are either coin symbol emojis or chat emojis.
+        assert!(
+            vector::all(&emoji_bytes, |emoji| {
+                is_a_supported_symbol_emoji(*emoji) ||
+                is_a_supported_chat_emoji(*emoji)
+            }),
+            E_NOT_SUPPORTED_CHAT_EMOJI,
+        );
+
+        // Construct the chat message from the emoji indices.
+        let message: String = string::utf8(b"");
+        vector::for_each(emoji_indices_sequence, |idx| {
+            let emoji = *vector::borrow(&emoji_bytes, (idx as u64));
+            string::append_utf8(&mut message, emoji);
+        });
+
+        assert!(
+            string::length(&message) <= (MAX_CHAT_MESSAGE_LENGTH as u64),
```

Comment:
This seems pretty restrictive, I estimate that the cap will be only 15/20 emojis in some cases?

How about capping the length of `emoji_indices_sequence` to `MAX_CHAT_MESSAGE_LENGTH`?

Reply:
Done!